### DISCUSSION TITLE
displacement-map-flag example

### DISCUSSION THREAD

**Discussion by ripmurdock** [pixijs/pixijs#9204] [Q&A]
Hi, I'm just getting started with pixijs, and also GSAP.

I was able to copy this example and place the app.view in a div:
https://pixijs.io/examples/#/filters-basic/displacement-map-flag.js

How do I set the height and width to match the PNG linked to the flag?
const flag = PIXI.Sprite.from('')
container.addChild(flag);

Also, is it possible to preserve the transparency in the PNG?

Thanks!

**Comment by SuperSodaSea** [comment]
The `Sprite` will automatically match the size of your PNG image.
Also the transparency of the PNG image is preserved without any extra effort. You can change the background color to verify this, for example change the first line of the example you give into `const app = new PIXI.Application({ background: 'blue' });`.

**Comment by ripmurdock** [comment]
Thanks!  Is there an asset hosting site that doesn't generate errors on
either jsfiddle or codepen when applying pixijs? I uploaded a couple of
test images to w3spaces.com. Both jsfiddle and codepen display images in
the results window, but the pixeljs div is blank and both jsfiddle or
codepen have similar errors in the console: Cross-Origin Request Blocked:
The Same Origin Policy disallows reading the remote resource at
fcd1.w3spaces.com/basketclip.png

It looks like both jsfiddle and codepen allow github assets to be used with
pixijs, and they will display assets from w3spaces.com, but they block any
pixijs transformations from being applied to assets hosted on w3spaces.com.
Any idea where I can upload assets for testing pixijs? Here the images
hosted on w3spaces.com are displayed, but pixijs is blocked from using
those images: jsfiddle.net/ripmurdock/examL49p/6
codepen.io/fcdobbs/pen/RwYKWr <https://codepen.io/fcdobbs/pen/RwYKWrj>

Thanks again!

**Reply by bigtimebuddy** [reply]
Try https://codesandbox.io/ instead. You can create a project and upload images.

**Comment by ripmurdock** [comment]
Super.  Thanks!

FYI I was stumped for two days in learning pixijs because there's a typo in
the pixijs code in this stackoverflow post:
https://stackoverflow.com/questions/71578674/canvas-pixi-js-background-transparent
